tiny_dnn/layers/average_pooling_layer.h

```diff
@@ -29,27 +29,28 @@ inline void tiny_average_pooling_kernel(
   float_t scale_factor,
   std::vector<typename partial_connected_layer::wi_connections> &out2wi) {
   for_i(parallelize, in_data[0]->size(), [&](size_t sample) {
-    const vec_t &in = (*in_data[0])[sample];
-    const vec_t &W  = (*in_data[1])[0];
-    const vec_t &b  = (*in_data[2])[0];
-    vec_t &out      = (*out_data[0])[sample];
+    const Tensor<> in((*in_data[0])[sample]);
+    const Tensor<> W((*in_data[1])[0]);
+    const Tensor<> b((*in_data[2])[0]);
+    Tensor<> out((*out_data[0])[sample]);
 
     auto oarea = out_dim.area();
     size_t idx = 0;
     for (size_t d = 0; d < out_dim.depth_; ++d) {
-      float_t weight = W[d] * scale_factor;
-      float_t bias   = b[d];
+      float_t weight = W.host_at(d) * scale_factor;
+      float_t bias   = b.host_at(d);
       for (size_t i = 0; i < oarea; ++i, ++idx) {
         const auto &connections = out2wi[idx];
-        float_t value{0};
-        for (auto connection : connections) value += in[connection.second];
+        float_t value = 0.0;
+        for (auto connection : connections) value += in.host_at(connection.second);
         value *= weight;
         value += bias;
-        out[idx] = value;
+        out.host_at(idx) = value;
       }
     }
 
     assert(out.size() == out2wi.size());
+    (*out_data[0])[sample] = out.toVec();
```

Comment:
it creates an instance of `vec_t` and returns the value
https://github.com/tiny-dnn/tiny-dnn/blob/feat/xtensor_integration/tiny_dnn/core/framework/tensor.h#L383-L388